_posts/2016-05-11-aws-lambdas.md

```diff
@@ -0,0 +1,152 @@
+---
+layout: post
+name: aws-lambdas 
+title: AWS Lambda for Beginners
+date: 2016-05-11 00:20:00 +00:00
+author: Mashooq Badar 
+image:
+   src: /assets/img/custom/blog/aws-lambda.png
+tags:
+- microservices 
+- devops 
+- aws
+---
+
+AWS Lambda is a compute service from Amazon. It makes deployment and provisioning very simple and fits very well with microservices based architecture. You can find out more about AWS Lambda [here](http://docs.aws.amazon.com/lambda/latest/dg/welcome.html). Currently supported platforms are JVM, Node JS and Python. 
+
+The programming model for the lambdas consists of **Handler, Context Object, Logging and Exceptions**. These are described [here](http://docs.aws.amazon.com/lambda/latest/dg/programming-model-v2.html). Lambda must not hold state because they are brought up and down and replicated as needed. Persistent state should be stored in a service that is outside the lifecycle of the lambda such as Amazon DynamoDB, S3 etc.
+
+First of all follow the instructions [here](http://docs.aws.amazon.com/lambda/latest/dg/setup.html) to setup an AWS Account and AWS Command-line Interface and note down you account id. 
```

Comment:
Working environment is irrelevant here. The linked instructions provide all the details.